crates/chain/tests/test_indexed_tx_graph.rs

```diff
@@ -206,86 +207,112 @@ fn test_list_owned_txouts() {
+                    }),
+            )
+        }),
         None,
     );
 
     let _ = graph.insert_relevant_txs([&tx4, &tx5].iter().map(|tx| (*tx, None)), Some(100));
 
     // A helper lambda to extract and filter data from the graph.
-    let fetch = |ht: u32, graph: &IndexedTxGraph<BlockId, KeychainTxOutIndex<String>>| {
-        let txouts = graph
-            .list_owned_txouts(&local_chain, local_chain.get_block(ht).unwrap())
-            .collect::<Vec<_>>();
-
-        let utxos = graph
-            .list_owned_unspents(&local_chain, local_chain.get_block(ht).unwrap())
-            .collect::<Vec<_>>();
-
-        let balance = graph.balance(
-            &local_chain,
-            local_chain.get_block(ht).unwrap(),
-            |spk: &Script| trusted_spks.contains(spk),
-        );
-
-        assert_eq!(txouts.len(), 5);
-        assert_eq!(utxos.len(), 4);
-
-        let confirmed_txouts_txid = txouts
-            .iter()
-            .filter_map(|full_txout| {
-                if matches!(full_txout.chain_position, ObservedAs::Confirmed(_)) {
-                    Some(full_txout.outpoint.txid)
-                } else {
-                    None
-                }
-            })
-            .collect::<BTreeSet<_>>();
-
-        let unconfirmed_txouts_txid = txouts
-            .iter()
-            .filter_map(|full_txout| {
-                if matches!(full_txout.chain_position, ObservedAs::Unconfirmed(_)) {
-                    Some(full_txout.outpoint.txid)
-                } else {
-                    None
-                }
-            })
-            .collect::<BTreeSet<_>>();
-
-        let confirmed_utxos_txid = utxos
-            .iter()
-            .filter_map(|full_txout| {
-                if matches!(full_txout.chain_position, ObservedAs::Confirmed(_)) {
-                    Some(full_txout.outpoint.txid)
-                } else {
-                    None
-                }
-            })
-            .collect::<BTreeSet<_>>();
-
-        let unconfirmed_utxos_txid = utxos
-            .iter()
-            .filter_map(|full_txout| {
-                if matches!(full_txout.chain_position, ObservedAs::Unconfirmed(_)) {
-                    Some(full_txout.outpoint.txid)
-                } else {
-                    None
-                }
-            })
-            .collect::<BTreeSet<_>>();
-
-        (
-            confirmed_txouts_txid,
-            unconfirmed_txouts_txid,
-            confirmed_utxos_txid,
-            unconfirmed_utxos_txid,
-            balance,
-        )
-    };
+    let fetch =
+        |height: u32,
+         graph: &IndexedTxGraph<ConfirmationHeightAnchor, KeychainTxOutIndex<String>>| {
+            let txouts = graph
+                .list_owned_txouts(
+                    &local_chain,
+                    local_chain
+                        .get_blockhash(height)
+                        .map(|hash| BlockId { height, hash })
+                        .unwrap(),
```

Comment:
nit: just save this in a `tip` variable instead of repeating it over and over again (unless it's changing between calls, but it doesn't seem so?)